### ISSUE TITLE
Unable to install with pipx, incompatible importlib-resources

### ISSUE THREAD

**Issue by bosd** [pre-commit/pre-commit#1490]
``pipx install pre-commit``

returns an error

```
ERROR: virtualenv 20.0.21 has requirement importlib-resources<2,>=1.0; python_version < "3.7", but you'll have importlib-resources 2.0.0 which is incompatible.
Internal error with venv metadata inspection.
```
How to resolve this?

steps to reproduce:
Ubuntu 18.04 host system
```
python3 -m pip install --user pipx
python3 -m pipx ensurepath
pipx install invoke
pipx install pre-commit
```

**Comment by DraXus** [comment]
I just had same issue. The problem is that `pre-commit` requires to install latest `virtualenv` and latest `importlib-resources`. Yesterday `importlib-resources` had a new release (from 1.5.0 to 2.0.0), which is not compatible with latest version of `virtualenv`. 

To fix this issue, just add `importlib-resources==1.5.0` (or < 2) to your requirements file just before pre-commit line.

**Comment by asottile** [comment]
`pre-commit` actually doesn't care about what version of `importlib-resources` to use, but `pip` doesn't yet know how to resolve diamond dependencies so it sees `imporltib-resources` first and resolves that to the latest, and then only later does it see the diamond via `virtualenv` but it's too late.  pip is growing a dependency resolver and will eventually be able to figure that out

I'd suggest you make an issue on `virtualenv` to become compatible with the latest version (it's likely that it already is and the `<` was over-zealous)

**Comment by bosd** [comment]
> I just had same issue. The problem is that `pre-commit` requires to install latest `virtualenv` and latest `importlib-resources`. Yesterday `importlib-resources` had a new release (from 1.5.0 to 2.0.0), which is not compatible with latest version of `virtualenv`.
> 
> To fix this issue, just add `importlib-resources==1.5.0` (or < 2) to your requirements file just before pre-commit line.

I am not sure how to do that.
Worked around it for now.
Ubuntu 18.04 comes shipped with Python 3.6.9
I upgraded it to Python 3.7.7.
That solved the issue
